packages/dd-trace/src/appsec/telemetry/rasp.js

```diff
@@ -26,25 +32,89 @@ function addRaspRequestMetrics (store, { duration, durationExt, wafTimeout, erro
   }
 }
 
-function trackRaspMetrics (metrics, raspRule) {
-  const tags = { rule_type: raspRule.type, waf_version: metrics.wafVersion }
+function trackRaspMetrics (store, metrics, raspRule) {
+  const versionsTags = getVersionsTags(metrics.wafVersion, metrics.rulesVersion)
+  const tags = { rule_type: raspRule.type, ...versionsTags }
 
   if (raspRule.variant) {
     tags.rule_variant = raspRule.variant
   }
 
+  if (metrics.wafVersion) {
+    store[DD_TELEMETRY_REQUEST_METRICS].wafVersion = metrics.wafVersion
+  }
+
+  if (metrics.rulesVersion) {
+    store[DD_TELEMETRY_REQUEST_METRICS].rulesVersion = metrics.rulesVersion
+  }
+
+  if (metrics.ruleTriggered) {
+    store[DD_TELEMETRY_REQUEST_METRICS].ruleTriggered = true
+  }
+
   appsecMetrics.count('rasp.rule.eval', tags).inc(1)
 
+  if (metrics.duration) {
+    appsecMetrics.distribution('rasp.rule.duration', tags).track(metrics.duration)
+
+    const raspDuration = store[DD_TELEMETRY_REQUEST_METRICS].raspDuration
+    appsecMetrics.distribution('rasp.duration', versionsTags).track(raspDuration)
+  }
+
+  if (metrics.durationExt) {
+    const raspDurationExt = store[DD_TELEMETRY_REQUEST_METRICS].raspDurationExt
+    appsecMetrics.distribution('rasp.duration_ext', versionsTags).track(raspDurationExt)
+  }
+
+  if (metrics.errorCode) {
+    const errorTags = { ...tags, waf_error: metrics.errorCode }
+
+    appsecMetrics.count('rasp.error', errorTags).inc(1)
+  }
+
   if (metrics.wafTimeout) {
     appsecMetrics.count('rasp.timeout', tags).inc(1)
   }
+}
+
+function trackRaspRuleMatch (store, raspRule, blockTriggered, blocked) {
+  if (!store[DD_TELEMETRY_REQUEST_METRICS].ruleTriggered) return
+
+  const tags = {
+    waf_version: store[DD_TELEMETRY_REQUEST_METRICS].wafVersion,
+    event_rules_version: store[DD_TELEMETRY_REQUEST_METRICS].rulesVersion,
+    rule_type: raspRule.type
+  }
 
-  if (metrics.ruleTriggered) {
-    appsecMetrics.count('rasp.rule.match', tags).inc(1)
+  if (raspRule.variant) {
+    tags.rule_variant = raspRule.variant
+  }
+
+  const block = getRuleMatchBlockingStatus(blockTriggered, blocked)
+  appsecMetrics.count('rasp.rule.match', { ...tags, block }).inc(1)
```

Comment:
why have spread operator here ??? just add `block: getRuleMatchBlockingStatus()` in the `tags` declaration no ?